test/utils/clonableGenerator.js

```diff
@@ -0,0 +1,156 @@
+import test from 'tape';
+import { spy } from 'sinon';
+import { cloneableGenerator } from '../../src/internal/utils';
+
+test('it should proxy the original generator', assert => {
+  assert.plan(4);
+
+  const spyGenerator = {
```

Comment:
technically this is an iterator, not a generator

definition of the generator is a function which produced an iterator object